packages/wonder-blocks-button/src/components/button.js

```diff
@@ -184,44 +167,40 @@ type Props =
            * If both safeWithNav and beforeNav are provided, beforeNav will be run
            * first and safeWithNav will only be run if beforeNav does not reject.
            */
-          beforeNav: () => Promise<mixed>,
-      |}
-    | {|
-          ...SharedProps,
-
-          href: string,
+          beforeNav?: () => Promise<mixed>,
 
           /**
            * Run async code in the background while client-side navigating. If the
            * browser does a full page load navigation, the callback promise must be
            * settled before the navigation will occur. Errors are ignored so that
            * navigation is guaranteed to succeed.
            */
-          safeWithNav: () => Promise<mixed>,
+          safeWithNav?: () => Promise<mixed>,
+
+          type?: empty,
       |}
     | {|
-          ...SharedProps,
-
-          href: string,
-
           /**
-           * Run async code before navigating. If the promise returned rejects then
-           * navigation will not occur.
-           *
-           * If both safeWithNav and beforeNav are provided, beforeNav will be run
-           * first and safeWithNav will only be run if beforeNav does not reject.
+           * Used for buttons within <form>s.
            */
-          beforeNav: () => Promise<mixed>,
+          type?: "submit",
 
-          /**
-           * Run async code in the background while client-side navigating. If the
-           * browser does a full page load navigation, the callback promise must be
-           * settled before the navigation will occur. Errors are ignored so that
-           * navigation is guaranteed to succeed.
-           */
-          safeWithNav: () => Promise<mixed>,
+          href?: empty, // can't be used with submit
+          beforeNav?: empty, // must be used with href
+          safeWithNav?: empty, // must be used with href
+          target?: empty, // must be used with href
+          rel?: empty, // must be used with href
```

Comment:
I've added inline comments here so that the comment shows up in the error message if Flow/TS finds any violations of these conditional props.